userspace/libsinsp/threadinfo.cpp

```diff
@@ -767,6 +768,44 @@ uint64_t sinsp_threadinfo::get_fd_limit()
 	return get_main_thread()->m_fdlimit;
 }
 
+void sinsp_threadinfo::traverse_parent_state(std::function<bool (sinsp_threadinfo *)> visitor)
+{
+	// Use two pointers starting at this, traversing the parent
+	// state, at different rates. If they ever equal each other
+	// before slow is NULL there's a loop.
+
+	sinsp_threadinfo *slow=this->get_parent_thread(), *fast=slow;
+
+	while(slow)
+	{
+		if(!visitor(slow))
+		{
+			break;
+		}
+
+		// If not at the end but fast == slow, there's a loop
+		// in the thread state.
+		if(slow == fast)
```

Comment:
fast should also be increased before the while, otherwise it will exit in the first iteration.